### ISSUE TITLE
Why is the exponent 32 bits?

### ISSUE THREAD

**Issue by vinniefalco** [ulfjack/ryu#110]
I see in the floating_decimal structure the exponent is stored using a 32-bit signed integer. But exponents for doubles only have a range of ~2,000 distinct integers, why isn't this a short instead?

**Comment by ulfjack** [comment]
No particular reason. I mostly write Java code, so don't expect my C to be top-notch. ;-)

**Comment by vinniefalco** [comment]
I think it deserves a comment, or to get demoted to the minor leagues as a short. Also, the algorithm for converting a double to a base-10 mantissa/exponent pair is generally useful.

**Comment by ulfjack** [comment]
Sure. Happy to merge a change if someone sends a PR. I can do it myself as well, but I'm pretty busy this week and next. (Someone needs to review the code to make sure that a short is safe there.)

**Comment by StephanTLavavej** [comment]
I can review such a PR. Would you want to narrow all types used for storing exponents, or just the ones in the struct? (In advance, I recommend keeping f2s and d2s aligned as much as possible.)

In general, I expect no performance differences between `int16_t` and `int32_t` outside of arrays, but I suppose it is conceptually cleaner to use narrower types when ranges are known, despite the usual arithmetic conversions/integral promotions widening tiny types back to `int` and so forth.

(There is a big performance difference between `int32_t` and `int64_t` for 32-bit platforms, which I've spent a surprising amount of time on.)

**Comment by vinniefalco** [comment]
> Would you want to narrow all types used for storing exponents, or just the ones in the struct?

The objective is to make `floating_decimal_64` a public type, and also make public the function which fills out the struct from the mantissa and base-2 exponent. Since the author has extensive changes planned, probably it is best to just change the type in the struct for now.

I am using only Visual Studio, and cmake, and I did not see any integration for getting a solution and project files up and running, so I am not able to properly test the effects of changing the type, thus I have submitted a pull request containing only the comment.

Thanks!
